packages/jaeger-ui/src/utils/readJsonFile.tsx

```diff
@@ -22,21 +38,36 @@ export default function readJsonFile(fileList: { file: File }): Promise<string>
         reject(new Error('Invalid result type'));
         return;
       }
+      let traceObj;
       try {
-        const traceObj = JSON.parse(reader.result);
-        if ('resourceSpans' in traceObj) {
-          JaegerAPI.transformOTLP(traceObj)
-            .then((result: string) => {
-              resolve(result);
-            })
-            .catch(() => {
-              reject(new Error(`Error converting traces to OTLP`));
-            });
-        } else {
-          resolve(traceObj);
+        traceObj = JSON.parse(reader.result);
+      } catch (error) {
+        try {
+          traceObj = tryParseMultiLineInput(reader.result);
+        } catch (error) {
+          reject(new Error(`Error processing JSON: ${(error as Error).message}`));
```

Comment:
`Error processing JSON` will be repeated twice. Maybe you don't need try/catch inside `tryParseMultiLineInput`?

Reply:
yeah, I was thinking about this. The thought process was that what if there was a newline seperated non-JSON entry? Like suppose the 3rd line in the file was the non json. In case we get rid of this try/catch inside tryParseMultiLineInput, the error will be propogated upwards, to the other catch. But, if in the future we do need a custom message to be displayed on this error, it wouldn't be possible without this.

Eg: "Error on L3" ...

Either ways it will work, I'm more inclined to keeping the try catch.

Reply:
yeah, but you are not giving any more precise error message in the inner try/catch, so it provides no value. If you want to add line number I think that would be more useful for troubleshooting. And I would change the error message from the inner try/catch to say something that parsing a single like failed.

Reply:
Yup, got it. I'll make the changes.